marlowe-marketplace-client/README.md

```diff
@@ -0,0 +1,113 @@
+
+```bash
+$ npm run start
+```
+
+The `start` script will:
+
+- Install npm dependencies
+- Compile the purescript code
+- Start the webkpack server
+
+Once the `start` script completes you can access the frontend via [http://localhost:8009](http://localhost:8009)
+
+## Development Workflow
+
+The following outlines some essentials for actually working on the marlowe marketplace code.
+
+### NPM Scripts
+
+Apart from the `start` script introduced above there are a couple of scripts for the most frequent tasks during development. In order to run a webpack server in development mode with automatic reloading use **webpack:server**:
+
+```
+$ npm webpack:server
+```
+
+Please refer to [package.json](./package.json) for the full set of provided scripts.
+
+
+### Managing Dependencies
+
+There are two relevant sources of dependencies that have to be handled and integrated with Nix separately: _Javascript(/npm)_ dependencies and _PureScript_ dependencies.
+
+#### Managing NPM Dependencies
+
+The Javascript dependencies are handled by npm in [package.json](./package.json) (and [package-lock.json](./package-lock.json) which
+is updated by npm automatically).
+
+The npm dependencies are integrated with Nix via [npmlock2nix](https://github.com/tweag/npmlock2nix) almost completely transparently. Any changes to the lockfile will be picked up npmlock2nix automatically during the nix build. No
+additional files have to be generated or maintained.
+
+##### NodeJS GitHub dependencies
+
+In general, npm dependencies are handled by npmlock2nix automatically and transparently. The one exception to this rule are
+GitHub dependencies. In order for these to work in restricted evaluation mode (which is what hydra uses) you have to specify
+the sha256 of the dependency you want to use in your `buildNodeModules`. For example:
+
+```
+buildNodeModules {
+    projectDir = ./.;
+    packageJson = ./package.json;
+    packageLockJson = ./package-lock.json;
+    githubSourceHashMap = {
+      shmish111.nearley-webpack-loader."939360f9d1bafa9019b6ff8739495c6c9101c4a1" = "1brx669dgsryakf7my00m25xdv7a02snbwzhzgc9ylmys4p8c10x";
+    };
+}
+```
+
+You can add new dependencies with the sha256 set to `"0000000000000000000000000000000000000000000000000000"`. This will yield an error
+message during the build with the actual hash value.
+
+#### Managing PureScript Dependencies
+
+The PureScript dependencies are handled by [spago](https://github.com/purescript/spago) in [packages.dhall](./packages.dhall).
+
+The Nix integration is done using [spago2nix](https://github.com/justinwoo/spago2nix). Any changes to the PureScript dependencies
+require an update of the Nix code generated by spago2nix:
+
+```
+$ spago2nix generate
+```
+
+This will parse the spago configuration and will generate an updated `.nix` file.
+
+**Note**: If the `spago2nix` command is not available make sure you are inside a nix-shell environment or that your lorri session
+is up and running.
+
+
+### Nix
```

Comment:
This part was copy-pasted from the dashboard and I haven't tested it yet. I will do it in a separate PR focused on deploying.